lib/main.js

```diff
@@ -55,13 +59,59 @@ export default {
           });
         }
       }),
-    );
-
-    this.subscriptions.add(
       atom.config.observe('linter-tslint.useLocalTslint', (use) => {
         tslintCache.clear();
         this.useLocalTslint = use;
       }),
+      atom.config.observe('linter-tslint.ignoreTypings', (ignoreTypings) => {
+        this.ignoreTypings = ignoreTypings;
+      }),
+    );
+
+    // Marks each TypeScript editor with a CSS class so that
+    // we can enable commands only for TypeScript editors.
+    this.subscriptions.add(
+      atom.workspace.observeTextEditors((textEditor) => {
+        if (textEditor.getRootScopeDescriptor().scopes.some(scope => scope === typescriptScope)) {
+          textEditor.element.classList.add(editorClass);
+        }
+      }),
+    );
+
+    // Command subscriptions
+    this.subscriptions.add(
+      atom.commands.add(`atom-text-editor.${editorClass}`, {
+        'linter-tslint:fix-file': async () => {
+          const textEditor = atom.workspace.getActiveTextEditor();
+
+          if (!textEditor || textEditor.isModified()) {
+            // Abort for invalid or unsaved text editors
+            atom.notifications.addError('Linter-TSLint: Please save before fixing');
+            return;
+          }
+
+          // The fix replaces the file content and the cursor jumps automatically
+          // to the beginning of the file, so save current cursor position
+          const cursorPosition = textEditor.getCursorBufferPosition();
```

Comment:
Just a note: This only happens _sometimes_, and when it does happen, restoring the cursor like this for some reason doesn't seem to work from the limited testing from https://github.com/AtomLinter/linter-eslint/pull/853 (it hasn't been pushed out to the public yet).